libsmtutil/SolverInterface.h

```diff
@@ -189,7 +189,7 @@ class Expression
 
 	static Expression ite(Expression _condition, Expression _trueValue, Expression _falseValue)
 	{
-		smtAssert(*_trueValue.sort == *_falseValue.sort, "");
+		smtAssert(_trueValue.sort->kind == _falseValue.sort->kind, "");
```

Comment:
```suggestion
		if (_trueValue.sort->kind == Kind::IntSort)
				smtAssert(_trueValue.sort->kind == _falseValue.sort->kind, "");
		else
				smtAssert(*_trueValue.sort == *_falseValue.sort, "");
```

Not sure if this would be better.

Reply:
Yea I think that's better. For BV and Array we definitely need to check the entire sort and not only the kind